src/ocean/THCM.C

```diff
@@ -331,6 +340,51 @@ THCM::THCM(Teuchos::ParameterList& params, Teuchos::RCP<Epetra_Comm> comm) :
 
     INFO("THCM init: m_global::initialize... done");
 
+    if (comm_->MyPID() == 0) {
```

Comment:
I'd prefer if these just existed on all processors I think.

Reply:
I was just following the original code, but I don't see a problem in running this on every node (if we do the same for all the preceding bits).

Reply:
Erik called that #legacy before. I think this currently works based on assumptions that are unclear. It'd be better to remove the need for those assumptions.